src/fhirtypes/common.ts

```diff
@@ -301,8 +321,17 @@ export function replaceReferences<T extends AssignmentRule | CaretValueRule>(
   } else if (value instanceof FshCode) {
     const [system, ...versionParts] = value.system?.split('|') ?? [];
     const version = versionParts.join('|');
-    const codeSystem = tank.fish(system, Type.CodeSystem);
-    const codeSystemMeta = fisher.fishForMetadata(codeSystem?.name, Type.CodeSystem);
+    let codeSystem = tank.fish(system, Type.CodeSystem);
+    let codeSystemMeta: Metadata;
+    if (codeSystem) {
+      codeSystemMeta = fisher.fishForMetadata(codeSystem.name, Type.CodeSystem);
+    } else {
+      const csInstance = tank.fish(system, Type.Instance);
+      if (csInstance && isInstance(csInstance) && csInstance.instanceOf === 'CodeSystem') {
```

Comment:
I notice this doesn't check that the `csInstance` is definitional (has `Usage: #definition`). This was something @cmoesel specified in the [original issue](https://github.com/FHIR/sushi/issues/801). I don't think it's likely someone would define a `CodeSystem` that is _not_ meant to be definitional, but maybe it is good to require they be explicit? Not sure, would like to hear @cmoesel's thoughts since he did the issue.

Reply:
Excellent point. Commit `081fd08` requires that the instance in question has a `usage` of `Definition` to be resolved as a `CodeSystem`.